### DISCUSSION TITLE
How to Batch Process : Confused about the basic usage of this extension

### DISCUSSION THREAD

**Discussion by jeekca** [glucauze/sd-webui-faceswaplab#55] [Guide/Doc]
If your image sequence wasn't generated by SD, you might want to steer clear of the Text2Img and Img2Img tabs. Your focus should be on the FaceSwapLab tab.

Within the FaceSwapLab environment, head straight to the Batch Process tab. Here, you'll notice sections labeled "Face 1" and "Source Images". Your reference image, the face you wish to superimpose onto your sequence, goes into "Face 1". Simply drag and drop it there. Next, your entire image sequence should be dragged and dropped into the "Source Images" section.

![Screenshot 2023-08-21 at 15-27-26 Stable Diffusion](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/8313c6d4-d16e-46a9-aac2-19419fa1328d)

**Original Message :** 

Hi!

I’m so sorry for asking such a basic/novice question. I am not new to SD, Automatic1111 or extensions, and yet, I can’t get FaceSwap to work despite reading the doc multiple times.

I guess one way to frame my question is: I have a reference image and an image sequence. How do I use the extension to swap the face found in my image sequence with the provided reference image?

Since my image sequence wasn’t generated by SD, I guess I shouldn’t use the extension within the Text2Img and Img2Img tabs, but use the FaceSwapLab tab? But in this tab, unless I’m missing something, there is no button to trigger the swapping (i.e. "Generate")?

Again, my apologies for asking such a basic question. Beyond this, the advanced settings make sense and I’m excited to batch-swap once I can get past this admittedly simple issue.

Thanks!

**Comment by jeekca** [comment]
Ah. My bad. I wish I had played around a tiny bit more before posting this.

I was right about using the FaceSwapLab > Batch Process tab exclusively, where I put a face in Face 1 and then drag/dropped my image sequence in the Source Images tab.

Closing this. Thanks!

**Reply by schoenid** [reply]
What confused me, was the possibility to add "Batch Source Images" to the "Face" tabs. (What is the purpose?)
Then I've tried to hold the whole "Face 1" tab with the mouse and tried to drag it over to the "Source Images" tab, but it's not possible. (... to drag the whole "image sequence", built from reference and source ...)

It took me some time to realize, that the "Batch Source Images" have not to be dragged in the "Face" tabs, but only in the "Source Images" tab. If only one image is used, it's not a sequence ...

But anyway, thanks for the hints with the red frames.

**Comment by AwesomeLemon** [comment]
I was confused about the same exact thing, thank you for posting the solution!

**Comment by waitwhatdidido** [comment]
im sorry im missing something here still. i have a pic in face one. ive also ran it in the tool and have it saved. ive clicked enable.

i have a pic in source images. i click generate and nothing.
